bot/exts/help_channels/_cog.py

```diff
@@ -178,8 +180,23 @@ async def get_available_candidate(self) -> discord.TextChannel:
                 if last_notification:
                     self.last_notification = last_notification
                     self.bot.stats.incr("help.out_of_channel_alerts")
-
                 channel = await self.wait_for_dormant_channel()
+        message = await channel.fetch_message(channel.last_message_id)
+        channels = _channel.get_category_channels(constants.Categories.help_available)
+        string = AVAILABLE_CHANNELS_MESSAGE.format(len(channels), ", ".join(channels))
+        # Attempt to get the message from channel.history if last_message_id fails.
+        if message is None:
+            history = await channel.history()
+            try:
+                message = await history.flatten()[0]
+            except IndexError:
+                pass
```

Comment:
Instead of flattening to a list, use the `next()` method of the iterator and catch the exception it raises.